### ISSUE TITLE
`~/.nixprofie` should be `~/.local/state/nix/profile` when `nix.settings.use-xdg-base-directories = true;`

### ISSUE THREAD

**Issue by Freed-Wu** [NixOS/nixpkgs#243826]
<https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/config/users-groups.nix#L683>

```nix
    environment.profiles = [
      "$HOME/.nix-profile"
      "/etc/profiles/per-user/$USER"
    ];
```

When `nix.settings.use-xdg-base-directories = true;`, it should be `$HOME/.local/state/nix/profile`.

**Comment by SuperSamus** [comment]
Should be `$XDG_STATE_HOME`, if it exists.

**Comment by Gerg-L** [comment]
My PR fixes this #241518
